modules/edit-modes/src/lib/draw-polygon-by-dragging-mode.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import type { ClickEvent, ModeProps } from '../types.js';
+import type { Polygon, FeatureCollection } from '../geojson-types.js';
+import { getPickedEditHandle } from '../utils.js';
+import { DrawPolygonMode } from './draw-polygon-mode';
+
+export class DrawPolygonByDraggingMode extends DrawPolygonMode {
+  handleClick() {
+    // No-op
+  }
+
+  handleStartDragging(event: ClickEvent) {
+    event.cancelPan();
+  }
+
+  handleStopDragging(event: ClickEvent, props: ModeProps<FeatureCollection>) {
+    this.addClickSequence(event);
+    const clickSequence = this.getClickSequence();
+
+    if (clickSequence.length > 2) {
+      // Complete the polygon.
+      const polygonToAdd: Polygon = {
+        type: 'Polygon',
+        coordinates: [[...clickSequence, clickSequence[0]]]
+      };
+
+      this.resetClickSequence();
+
+      const editAction = this.getAddFeatureOrBooleanPolygonAction(polygonToAdd, props);
+      if (editAction) {
+        props.onEdit(editAction);
+      }
+    }
+  }
+
+  handleDragging(event: ClickEvent) {
+    const { picks } = event;
+    const clickedEditHandle = getPickedEditHandle(picks);
+
+    if (!clickedEditHandle) {
```

Comment:
You may want to have some additional throttling logic here so you don't add thousands f points unnecessarily. Currently, points will be added as fast as the browser delivers pointer move events (and it doesn't overlap an existing point). But that may be more fidelity than some users need.

It could be a configuration that users can put in `modeConfig`. Something like `modeConfig.throttleMs` whose value is the number of milliseconds to wait after adding a point before adding another point. https://www.npmjs.com/package/lodash.throttle can be used to accomplish the throttling.